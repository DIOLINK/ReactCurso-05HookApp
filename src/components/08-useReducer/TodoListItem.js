import React from 'react';
import PropTypes from 'prop-types';

const TodoListItem = ({ todo, index, handleToggle, handleDelete }) => {
  return (
    <li className="list-group-item">
      <p
        className={`${todo.done && 'complete'}`}
        onClick={() => handleToggle(todo.id)}
      >
        {index + 1}.{todo.description}
      </p>
      <button className="btn btn-danger" onClick={() => handleDelete(todo.id)}>
        Delete
      </button>
    </li>
  );
};

TodoListItem.propTypes = {
  todo: PropTypes.object.isRequired,
  index: PropTypes.number.isRequired,
  handleToggle: PropTypes.func.isRequired,
  handleDelete: PropTypes.func.isRequired,
};

export default TodoListItem;
