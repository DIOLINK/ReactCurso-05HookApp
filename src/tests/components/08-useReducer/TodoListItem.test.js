import { shallow } from 'enzyme';
import TodoListItem from '../../../components/08-useReducer/TodoListItem';
import { dummyTodo } from '../../fixtures/dummyTodo';

describe('Probando <TodoListItem />', () => {
  const handleDelete = jest.fn();
  const handleToggle = jest.fn();
  const wrapper = shallow(
    <TodoListItem
      todo={dummyTodo[0]}
      index={0}
      handleDelete={handleDelete}
      handleToggle={handleToggle}
    />
  );
  it('Debe de mostrar el componente', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('Debe de llamar a la funcion handleDelete', () => {
    wrapper.find('button').simulate('click');
    expect(handleDelete).toHaveBeenCalledWith(dummyTodo[0].id);
  });
  it('Debe de llamar a la funcion handleToggle', () => {
    wrapper.find('p').simulate('click');
    expect(handleToggle).toHaveBeenCalledWith(dummyTodo[0].id);
  });
  it('Debe mostrar el texto del TODO', () => {
    expect(wrapper.find('p').text()).toBe('1.Hacer la compra');
  });
});
