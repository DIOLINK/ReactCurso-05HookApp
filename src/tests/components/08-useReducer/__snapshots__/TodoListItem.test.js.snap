// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando <TodoListItem /> Debe de mostrar el componente 1`] = `
<li
  className="list-group-item"
>
  <p
    className="false"
    onClick={[Function]}
  >
    1
    .
    Hacer la compra
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</li>
`;
